typing/typecore.ml

```diff
@@ -2925,6 +2969,8 @@ let collect_apply_args env funct ignore_labels ty_fun ty_fun0 sargs =
       | Tarrow (l, ty_arg, ty_ret, com), Tarrow (_, ty_arg0, ty_ret0, _)
         when is_commu_ok com ->
           Some (l, `Arrow (ty_arg, ty_ret, ty_arg0, ty_ret0))
+      | Tfunctor (l, id, pfl, t), Tfunctor (_, id0, pfl0, t0) ->
+          Some (l, `Functor (id, pfl, t, id0, pfl0, t0))
```

Comment:
This is too many parameters in a non-labeled tuples. We should have a record type or at least labels to indicate what the fields are. (My impression is that this could be structured as two things where one is `(id, pfl, t)` and the other `(id0, pfl0, t0)`.)